### ISSUE TITLE
I'm lost

### ISSUE THREAD

**Issue by zerocool206** [nsc-cs/BookStore#3]
This github thing is confusing. Pretty sure I deleted someones code also.............my bad guys.

**Comment by lulumialu** [comment]
No worries! How can I help?

**Comment by zerocool206** [comment]
My pushing isn't working correctly. I'm probably also going to have to talk to Matt. We haven't gone over arraylist yet, so I'm a bit confused on certain portions of my class.










On Wed, Feb 25, 2015 at 10:03 AM -0800, "Lauren" <notifications@github.com> wrote:
No worries! How can I help?

---

**Comment by mstaehely** [comment]
I'm happy to help whenever I am not in class or busy :) ArrayLists are my
second favorite data structure.

**Comment by lulumialu** [comment]
It's awesome that you know the ranking of your favorite data structures. XD What's the first?

**Comment by mstaehely** [comment]
I'm really partial to linked lists, actually. I think that conceptually
they're quite nice and they feel more 'clean'. Outweighs the indexing
problem, assuming the size is small enough. Right tool for the right job
and all that!

**Comment by mstaehely** [comment]
Also I'm on my way home now so if you have a question about ArrayLists, hit
me up whenever :)

**Comment by zerocool206** [comment]
Yeah, I mean I could read up on them. Overall, I think I'm just a bit confused by the working of the super class. We are now just going over this, and haven't looked at abstract yet. I

Date: Wed, 25 Feb 2015 13:00:15 -0800

**Comment by mstaehely** [comment]
No problem- give me a bit and I'll be on gchat if you'd like to reach me
there. Abstract methods are pretty straight forward, though.

**Comment by mstaehely** [comment]
Okay so- a quick primer on an abstract class.

An abstract class is one which has abstract methods. These are methods
which have a signature like a normal method, but no code associated with
them. It's the job of a subclass to implement those methods and put code in
them. To do that, your method signature has to match the abstract one
(minus the key word 'abstract'). So in the case of the getPeople() method,
you'd have to write 'public ArrayList<String> getPeople(){ //your code here
}'.

Abstract methods are very powerful tools, because they give you a
generalized requirement for all your subclasses.

A sub class has access to all the public members of its superclass. Also
I'm home again and on gchat if you have any questions. I'm happy to help!

**Comment by lulumialu** [comment]
I'm sure you've already looked at the docs, but just in case: http://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html
